### ISSUE TITLE
Suggestion - Tow Job Impound Price Adjustments

### ISSUE THREAD

**Issue by MetaHatter** [InZidiuZ/op-framework-issue-tracker#2614]
### Assurance

I have read the `README.md`

### Summary

Adjust impound prices to make tow job more appealing and focus target on city clean up.

Ideally utilize a query on the plate/vin to determine vehicle ownership at time of impound and compare against current player list.

- If local car : scale the price from $50-150 based on distance of the tow. (as the crow flies from pickup point to nearest impound lot)
- If citizen car and the citizen is still in city : Pay $250
- If citizen car and the owner is not in city : Pay $450
- If PD/EMS vehicle : Pay $0 

### Reason

Currently the conversion of citizen cars to local cars when a player flies out of the city forces tow drivers to focus on towing cars that are likely in use.

It would make more sense from an RP perspective as well as improve tow pay if they were rewarded for cleaning up abandoned vehicles.

This change would also incentivize people to park their cars at a garage or their homes when they fly out, increasing the need for real estate.

**Comment by CloudTheWolf** [comment]
I have to ask why PD/EMS would be $0?
This would decentivize tows wanting to clean-up these vehicles and just encourage pd to just use `/impound`  rather than call a tow to destroyed vehicles.

As for the Citizen car, I don't see why the player being in the city or not should impact the "Insurance Payout" (As I call it)

**Comment by MetaHatter** [comment]
> I have to ask why PD/EMS would be $0? This would decentivize tows wanting to clean-up these vehicles and just encourage pd to just use `/impound` rather than call a tow to destroyed vehicles.

PD/EMS can tip when you are asked to pick those up.  Even under the worst-case scenario, the dramatic increase in other opportunities afforded by this change would make it still a rather large net win.  

PD and EMS being available as often as possible is a server-wide benefit. Having them off the street tracking down a vehicle that, in reality, would never be towed without explicit permission, negatively impacts other people's RP.  Not worth the couple hundred dollars for the one tow truck driver.  

TLDR, this change would make the tow driver thousands of dollars more per day, so it won't matter.

**Comment by coalaura** [comment]
I dont think the player being in the city or not should matter at all to the impound money received. PD/EMS vehicles being $0 i dont like either.

> Ideally utilize a query on the plate/vin to determine vehicle ownership at time of impound and compare against current player list.

We don't need to use the plate of the vehicle since that can be faked or changed? We have states that keep track of the vehicle id.

> Currently the conversion of citizen cars to local cars when a player flies out of the city forces tow drivers to focus on towing cars that are likely in use.

Is this currently a thing? Do you get less money if the player flew out right now? Cause if yes thats a bug and not intentional

**Comment by MetaHatter** [comment]
> I dont think the player being in the city or not should matter at all to the impound money received. PD/EMS vehicles being $0 i dont like either.
> 
> > Ideally utilize a query on the plate/vin to determine vehicle ownership at time of impound and compare against current player list.
> 
> We don't need to use the plate of the vehicle since that can be faked or changed? We have states that keep track of the vehicle id.
> 
> > Currently the conversion of citizen cars to local cars when a player flies out of the city forces tow drivers to focus on towing cars that are likely in use.
> 
> Is this currently a thing? Do you get less money if the player flew out right now? Cause if yes thats a bug and not intentional

This was a thing the last time I checked (and the biggest issue for two revenue imho, the others above are FAR less important).  It may no longer be the case, but since I don't tow anymore I do not know for certain.

**Comment by vol5** [comment]
> Is this currently a thing? Do you get less money if the player flew out right now? Cause if yes thats a bug and not intentional

It converts the car to an npc car according to the tow system and gives $150 if I remember right, its been a very long standing bug.

**Comment by coalaura** [comment]
Have either of you 2 tested this very recently? I cant really test it myself locally rn but looking at the code there is nothing that changes the vehicle when the player flies out

**Comment by King2704** [comment]
Additionally, it would be nice if the trash trucks and the box trucks from the delivery jobs would pay the same amount as player owned vehicles when turning them in the impound.

**Comment by MetaHatter** [comment]
> Additionally, it would be nice if the trash trucks and the box trucks from the delivery jobs would pay the same amount as player owned vehicles when turning them in the impound.

People would abuse the ever-loving heck out of this....  Have friends spawn trucks, tow truck... friends cancel jobs, spawns new trucks... tow trucks... Free, endless monies.

**Comment by King2704** [comment]
The Waste collector and Delivery job could get a 15min cooldown added if the job was canceled to prevent them from doing that. 
(The idea behind it was mainly that tow truck drivers clean the streets of abandoned trucks due to people getting used as hostages and never returning to the vehicle)

**Comment by coalaura** [comment]
I'll be closing this soon since nobody responded to this https://github.com/InZidiuZ/op-framework-issue-tracker/issues/2614#issuecomment-1368587059 i am guessing its fixed

**Comment by coalaura** [comment]
marking this as completed with the recent changes
